packages/jest-cli/src/watch.js

```diff
@@ -379,6 +404,55 @@ export default function watch(
   return Promise.resolve();
 }
 
+const checkForConflicts = (watchPluginKeys, plugin, globalConfig) => {
+  const key = getPluginKey(plugin, globalConfig);
+  if (!key) {
+    return;
+  }
+
+  const conflictor = watchPluginKeys.get(key);
+
+  if (!conflictor || conflictor.overwritable) {
+    watchPluginKeys.set(key, {
+      overwritable: false,
+      plugin,
+    });
+    return;
+  }
+
+  let error;
+  if (conflictor.forbiddenOverwriteMessage) {
+    error = `Jest configuration error: watch plugin ${getPluginIdentifier(
+      plugin,
+    )} attempted to register key <${key}>, that is reserved internally for ${
+      conflictor.forbiddenOverwriteMessage
+    }.   Please change the configuration key for this plugin.`;
+  } else {
+    const plugins = [conflictor.plugin, plugin]
+      .map(getPluginIdentifier)
+      .join(' and ');
+    error = `Jest configuration error: watch plugins ${plugins} both attempted to register key <${key}>. Please change the key configuration for one of the conflicting plugins to avoid overlap.`;
+  }
+  console.error('\n\n' + chalk.red(error));
+  exit(64); // EX_USAGE
```

Comment:
I wonder if it makes more sense to just throw a `ValidationError` like we do for other configuration errors.

E.g.

```json
{
	"jest": {
	  "collectCoverage": "wat"
	}
}
```

![image](https://user-images.githubusercontent.com/1404810/43196685-1bf695c0-9009-11e8-8259-886d00dfaa79.png)

Reply:
@SimenB this is an interesting idea. Am I correct in assuming the right place for this would be in `jest-config`’s `normalize.js` file, in its existing handling of the `watchPlugins` option?

Reply:
(I'm not entirely sure we'd have sufficient/relevant info to populate the arguments to the plugin's `getUsageInfo()` there, I'll check. At the very least we can check inline key configs, though.)

Reply:
Yeah, in normalize seems natural, imo. Not sure if we have enough information there, but if not we can just hard-code the reserved ones